From a7e545674a8db54bf6cb7e04ef1147ea2c0baec8 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Wed, 3 Jan 2024 11:14:09 +0530
Subject: [PATCH 2/2] wifi: ath12k: update mac80211's perf_mode flag in
 driver

perf_mode flag is used in mac80211 to take performance boost path.
Update mac80211 ieee80211_hw struct's flag, when user disable stats in
driver.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/debugfs.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/ath/ath12k/debugfs.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs.c
@@ -2223,6 +2223,7 @@ static ssize_t ath12k_write_stats_disabl
 			pdev = &ab->pdevs[i];
 			if (pdev && pdev->ar) {
 				ath12k_debug_config_mon_status(pdev->ar, !disable);
+				pdev->ar->ah->hw->perf_mode = disable;
 
 				if (!disable)
 					mask = HTT_PPDU_STATS_TAG_DEFAULT;
